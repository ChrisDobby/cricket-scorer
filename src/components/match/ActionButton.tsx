import * as React from 'react';

const buttonStyle: React.CSSProperties = {
    minWidth: '36px',
    margin: '2px',
};

export interface ActionButtonProps {
    caption: string;
    noBall: boolean;
    action: () => void;
}

export const ActionButton = ({ caption, noBall, action }: ActionButtonProps) => (
    <button
        className={`btn ${noBall ? 'btn-danger' : 'btn-success'}`}
        style={buttonStyle}
        onClick={action}
    >{caption}
    </button>
);
