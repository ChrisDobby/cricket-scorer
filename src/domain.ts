export interface Profile {
    id: string;
    name: string;
    picture?: string;
}

export enum Howout {
    Bowled,
    Lbw,
    Caught,
    RunOut,
    Stumped,
    HitWicket,
    TimedOut,
    HandledBall,
    ObstructingField,
}

export enum UnavailableReason {
    Retired,
    Absent,
}

export enum DeliveryOutcome {
    Valid,
    Noball,
    Wide,
}

export enum MatchType {
    LimitedOvers,
    Time,
}

export enum InningsStatus {
    InProgress,
    AllOut,
    OversComplete,
    Declared,
    MatchComplete,
}

export enum TeamType {
    HomeTeam,
    AwayTeam,
}

export enum Result {
    HomeWin,
    AwayWin,
    Tie,
    Draw,
    Abandoned,
}

export enum WinBy {
    Runs,
    Wickets,
}

export enum BreakType {
    BadLight,
    CloseOfPlay,
    Innings,
    Lunch,
    Rain,
    Tea,
}

export enum EventType {
    Delivery,
    NonDeliveryWicket,
    BatterUnavailable,
    BatterAvailable,
    OverComplete,
}

export interface Wicket {
    time: number;
    howOut: Howout;
    bowlerIndex?: number;
    fielderIndex?: number;
}

export interface DeliveryScores {
    runs?: number;
    boundaries?: number;
    byes?: number;
    legByes?: number;
    wides?: number;
}

export interface DeliveryWicket {
    howOut: Howout;
    fielderIndex?: number;
    changedEnds: boolean;
}

export interface Outcome {
    deliveryOutcome: DeliveryOutcome;
    scores: DeliveryScores;
    wicket?: DeliveryWicket;
}

export interface Event {
    time: number;
    type: EventType;
}

export interface Delivery extends Event {
    bowlerIndex: number;
    batsmanIndex: number;
    overNumber: number;
    outcome: Outcome;
}

export interface NonDeliveryWicket extends Event {
    batsmanIndex: number;
    out: Howout;
}

export interface BatterUnavailable extends Event {
    batsmanIndex: number;
    reason: UnavailableReason;
}

export interface OverComplete extends Event {
    bowlerIndex: number;
    batsmanIndex: number;
}

export interface Over {
    deliveries: Delivery[];
    bowlingRuns: number;
    wickets: number;
}

export interface BattingInnings {
    runs: number;
    timeIn: number;
    ballsFaced: number;
    fours: number;
    sixes: number;
    wicket?: Wicket;
}

export interface Batter {
    playerIndex: number;
    innings?: BattingInnings;
    unavailableReason?: UnavailableReason;
}

export interface Bowler {
    playerIndex: number;
    completedOvers: number;
    totalOvers: string;
    maidenOvers: number;
    runs: number;
    wickets: number;
}

export interface Extras {
    byes: number;
    legByes: number;
    wides: number;
    noBalls: number;
    penaltyRuns: number;
}

export interface Batting {
    batters: Batter[];
    extras: Extras;
}

export interface FallOfWicket {
    wicket: number;
    batterIndex: number;
    score: number;
    partnership: number;
}

export interface Innings {
    status: InningsStatus;
    battingTeam: TeamType;
    bowlingTeam: TeamType;
    score: number;
    wickets: number;
    completedOvers: number;
    totalOvers: string;
    events: Event[];
    batting: Batting;
    bowlers: Bowler[];
    fallOfWickets: FallOfWicket[];
    maximumOvers?: number;
    completeTime?: number;
}

export interface Team {
    name: string;
    players: string[];
}

export interface MatchConfig {
    playersPerSide: number;
    type: MatchType;
    oversPerSide?: number;
    inningsPerSide: number;
    runsForNoBall: number;
    runsForWide: number;
}

export interface Toss {
    tossWonBy: TeamType;
    battingFirst: TeamType;
}

export interface MatchResult {
    result: Result;
    winBy?: WinBy;
    winMargin?: string;
}

export interface MatchBreak {
    type: BreakType;
    startTime: number;
    endTime?: number;
}

export interface Match {
    id?: string;
    user: string;
    config: MatchConfig;
    homeTeam: Team;
    awayTeam: Team;
    date: string;
    complete: boolean;
    status: string;
    toss?: Toss;
    result?: MatchResult;
    innings: Innings[];
    breaks: MatchBreak[];
}

export interface StoredMatch {
    match: Match;
    currentBatterIndex?: number;
    currentBowlerIndex?: number;
    version: number;
    lastEvent?: string;
}

interface MatchBeingScored {
    date: string;
    status: string;
    user: string;
    homeTeam: string;
    awayTeam: string;
    lastEvent: string | undefined;
    version: number;
}

export interface PersistedMatch extends MatchBeingScored {
    id: string;
}

export interface CurrentEditingMatch extends MatchBeingScored {
    id: string | undefined;
}

export interface InProgressMatch {
    match: Match;
    currentInnings?: Innings;
    currentBatter?: Batter;
    currentBowler?: Bowler;
    previousBowler?: Bowler;
    previousBowlerFromEnd?: Bowler;
    currentOver?: Over;
    currentOverComplete?: boolean;
    provisionalInningsStatus?: InningsStatus;
    provisionalMatchComplete: boolean;
    nextBattingTeam?: Team;
    canSelectBattingTeamForInnings: boolean;
    newBatterRequired: boolean;
    version: number;
    lastEvent?: string;
    startMatch: (tossWonBy: TeamType, battingFirst: TeamType) => void;
    startInnings: (battingTeam: TeamType, batter1Index: number, batter2Index: number, overs?: number) => void;
    newBowler: (playerIndex: number) => void;
    newBatter: (playerIndex: number) => void;
    delivery: (deliveryOutcome: DeliveryOutcome, scores: DeliveryScores, wicket: DeliveryWicket | undefined) => void;
    undoPreviousDelivery: () => void;
    completeOver: () => void;
    flipBatters: () => void;
    completeInnings: (status: InningsStatus) => void;
    completeMatch: (result: MatchResult) => void;
    nonDeliveryWicket: (howout: Howout) => void;
    batterUnavailable: (playerIndex: number, reason: UnavailableReason) => void;
    setFromStoredMatch: (storedMatch: StoredMatch | undefined) => void;
    changeOrders: (battingOrder: number[], bowlingOrder: number[]) => void;
    rolledBackInnings: (eventIndex: number) => Innings | undefined;
    rollback: (eventIndex: number) => void;
    updateOvers: (overs: number) => void;
    startBreak: (breakType: BreakType) => void;
    undoToss: () => void;
    updateTeams: (homeTeam: string, awayTeam: string, homePlayers: string[], awayPlayers: string[]) => void;
    changeBowler: (fromDelivery: number, playerIndex: number) => void;
}

export interface RebuiltInnings {
    innings: Innings;
    batterIndex: number;
}

export interface UserTeams {
    user: string;
    teams: Team[];
}

export const howOutDescription = (
    getBowlerAtIndex: (index: number) => string,
    getFielderAtIndex: (index: number) => string,
    sameBowlerAndFielder: (bowlerIndex: number, fielderIndex: number) => boolean,
) => (wicket?: Wicket): string => {
    const description = (wkt: Wicket): string => {
        switch (wkt.howOut) {
            case Howout.Bowled:
                return `bowled ${getBowlerAtIndex(<number>wkt.bowlerIndex)}`;
            case Howout.Lbw:
                return `lbw ${getBowlerAtIndex(<number>wkt.bowlerIndex)}`;
            case Howout.Caught:
                return sameBowlerAndFielder(<number>wkt.bowlerIndex, <number>wkt.fielderIndex)
                    ? `ct & bowled ${getBowlerAtIndex(<number>wkt.bowlerIndex)}`
                    : `ct ${getFielderAtIndex(<number>wkt.fielderIndex)} b ${getBowlerAtIndex(<number>(
                          wkt.bowlerIndex
                      ))}`;
            case Howout.RunOut:
                return wkt.fielderIndex ? `run out (${getFielderAtIndex(<number>wkt.fielderIndex)})` : 'run out';
            case Howout.Stumped:
                return `st ${getFielderAtIndex(<number>wkt.fielderIndex)} b ${getBowlerAtIndex(<number>(
                    wkt.bowlerIndex
                ))}`;
            case Howout.TimedOut:
                return 'timed out';
            case Howout.ObstructingField:
                return 'obstructing the field';
            case Howout.HandledBall:
                return 'handled the ball';
            case Howout.HitWicket:
                return `hit wkt ${getBowlerAtIndex(<number>wkt.bowlerIndex)}`;
        }
    };

    return wicket ? description(wicket) : 'not out';
};

export const unavailablDescription = (reason: UnavailableReason): string => {
    switch (reason) {
        case UnavailableReason.Absent:
            return 'absent';
        case UnavailableReason.Retired:
            return 'retired';
        default:
            return '';
    }
};

export const validDelivery = (delivery: Delivery) =>
    delivery.outcome.deliveryOutcome !== DeliveryOutcome.Noball &&
    delivery.outcome.deliveryOutcome !== DeliveryOutcome.Wide;

export const oversDescription = (completedOvers: number, currentOver: Delivery[]): string => {
    const validCurrentOver = currentOver.filter(validDelivery);
    const remainder = validCurrentOver.length > 0 ? `.${validCurrentOver.length}` : '';
    return `${completedOvers}${remainder}`;
};

const nonStrikerHowouts = [Howout.RunOut, Howout.TimedOut, Howout.HandledBall, Howout.ObstructingField];

export const howouts = (currentBatter: Batter) => (batter: Batter) =>
    Object.keys(Howout)
        .filter(key => !isNaN(Number(Howout[key])))
        .map(key => Howout[key])
        .filter(
            howout =>
                batter.playerIndex === currentBatter.playerIndex ||
                nonStrikerHowouts.find(how => how === howout) !== undefined,
        );

export const howoutRequiresFielder = (howout: Howout) =>
    howout === Howout.Caught || howout === Howout.Stumped || howout === Howout.RunOut;

export const howoutBattersCouldCross = (howout: Howout) => howout === Howout.Caught || howout === Howout.RunOut;

export const howoutCouldScoreRuns = (howout: Howout) =>
    howout === Howout.Caught || howout === Howout.RunOut || howout === Howout.ObstructingField;

export const howoutCouldBeNoBall = (howout: Howout) => howout === Howout.RunOut || howout === Howout.ObstructingField;

export const howoutCouldBeWide = (howout: Howout) =>
    howout === Howout.RunOut || howout === Howout.Stumped || howout === Howout.ObstructingField;

export const deliveries = (events: Event[]) =>
    events.filter(event => event.type === EventType.Delivery).map(event => event as Delivery);
