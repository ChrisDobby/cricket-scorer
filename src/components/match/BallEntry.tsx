import * as React from 'react';
import * as globalStyles from '../styles';
import { Innings, Batter, Bowler, oversDescription, BattingInnings } from '../../domain';
import { EntryPanel, BallFunctions } from './EntryPanel';

const battingScoreStyle: React.CSSProperties = {
    display: 'inline',
    marginRight: '10px',
};

export interface BallEntryProps {
    innings: Innings;
    batter: Batter;
    bowler: Bowler;
    ballFunctions: BallFunctions;
}

export const BallEntry = ({ innings, batter, bowler, ballFunctions }: BallEntryProps) => (
    <div style={globalStyles.sectionContainer}>
        <div className="row" style={globalStyles.singleHeadingRow}>
            <h4>{`${bowler.name} to ${batter.name}`}</h4>
        </div>
        <div className="row">
            <div className="col-12 col-lg-6">
                <div className="row">
                    <div className="col-6">
                        <h5>{innings.battingTeam.name}</h5>
                    </div>
                    <div className="col-6">
                        <h5>{`${innings.score}-${innings.wickets}`}</h5>
                    </div>
                </div>
                <div className="row">
                    <div className="col-6">
                        <h5>Overs</h5>
                    </div>
                    <div className="col-6">
                        <h5>{oversDescription(innings.completedOvers, [])}</h5>
                    </div>
                </div>
                {innings.batting.batters.filter(batter => batter.innings && !batter.innings.wicket)
                    .map((batter, idx) => ({
                        name: batter.name,
                        innings: batter.innings as BattingInnings,
                        index: idx,
                    }))
                    .map(currentBatter => (
                        <div key={currentBatter.index} className="row">
                            <div className="col-6">
                                <h5>{currentBatter.name}</h5>
                            </div>
                            <div className="col-6">
                                <span>
                                    <h5 style={battingScoreStyle}>{`${currentBatter.innings.runs}` +
                                        `(${currentBatter.innings.ballsFaced})`}</h5>
                                    {currentBatter.name === batter.name && <i className="fa fa-circle" />}
                                </span>
                            </div>
                        </div>
                    ))}
                <div className="row">
                    <div className="col-6">
                        <h5>{bowler.name}</h5>
                    </div>
                    <div className="col-6">
                        <h5>
                            {`${oversDescription(bowler.completedOvers, [])}-` +
                                `${bowler.maidenOvers}-${bowler.runs}-${bowler.wickets}`}
                        </h5>
                    </div>
                </div>
            </div>
            <div className="col-12 col-lg-6">
                <EntryPanel ballFunctions={ballFunctions} />
            </div>
        </div>
    </div>
);
