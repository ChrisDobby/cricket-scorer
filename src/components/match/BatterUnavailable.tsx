import * as React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogActions from '@material-ui/core/DialogActions';
import Button from '@material-ui/core/Button';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import { UnavailableReason, Innings } from '../../domain';

interface BatterUnavailableProps {
    reason: UnavailableReason;
    innings: Innings;
    battingPlayers: string[];
    update: (playerIndex: number, reason: UnavailableReason) => void;
    cancel: () => void;
}

export default (props: BatterUnavailableProps) => {
    const [playerIndex, setPlayerIndex] = React.useState(undefined as number | undefined);

    const batters = props.innings.batting.batters.filter(
        batter =>
            (props.reason === UnavailableReason.Absent && !batter.innings) ||
            (props.reason === UnavailableReason.Retired && batter.innings && !batter.innings.wicket),
    );
    const description = props.reason === UnavailableReason.Retired ? 'retired' : 'absent';

    return (
        <div>
            <Dialog open={true} aria-labelledby="update-innings-overs-title">
                <DialogTitle id="update-innings-overs-title">{`Mark one of the batting side as ${description}`}</DialogTitle>
                <DialogContent>
                    <DialogContentText>{`Select the ${description} player`}</DialogContentText>
                </DialogContent>
                <DialogContent>
                    {batters.length > 0 && (
                        <Select value={playerIndex} onChange={ev => setPlayerIndex(Number(ev.target.value))} fullWidth>
                            {batters.map(batter => (
                                <MenuItem key={batter.playerIndex} value={batter.playerIndex}>
                                    {props.battingPlayers[batter.playerIndex]}
                                </MenuItem>
                            ))}
                        </Select>
                    )}
                    {batters.length === 0 && (
                        <DialogContentText>{`No players can be marked as ${description}`}</DialogContentText>
                    )}
                </DialogContent>
                {batters.length > 0 && (
                    <DialogActions>
                        <Button
                            onClick={() => props.update(playerIndex as number, props.reason)}
                            color="primary"
                            autoFocus
                            disabled={typeof playerIndex === 'undefined'}
                        >
                            OK
                        </Button>
                        <Button onClick={props.cancel} color="primary">
                            Cancel
                        </Button>
                    </DialogActions>
                )}
                {batters.length === 0 && (
                    <DialogActions>
                        <Button onClick={props.cancel} color="primary" autoFocus>
                            OK
                        </Button>
                    </DialogActions>
                )}
            </Dialog>
        </div>
    );
};
