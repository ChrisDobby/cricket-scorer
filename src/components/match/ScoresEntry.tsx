import * as React from 'react';
import { DeliveryOutcome, DeliveryScores } from '../../domain';
import { ActionButton } from './ActionButton';
import { OtherScore } from './OtherScore';

const spacerStyle: React.CSSProperties = {
    width: '40px',
    display: 'inline-block',
};

export interface ScoresEntryProps {
    showDot: boolean;
    deliveryOutcome: DeliveryOutcome;
    getScores: (score: number) => DeliveryScores;
    action: (deliveryOutcome: DeliveryOutcome, scores: DeliveryScores) => void;
}

export const ScoresEntry = ({ showDot, deliveryOutcome, getScores, action }: ScoresEntryProps) => {
    const noBall = deliveryOutcome === DeliveryOutcome.Noball;
    return (
        <div className="col-10">
            {showDot &&
                <ActionButton caption="." action={() => action(deliveryOutcome, getScores(0))} noBall={noBall} />}
            {!showDot &&
                <span style={spacerStyle}/>}
            <ActionButton caption="1" action={() => action(deliveryOutcome, getScores(1))} noBall={noBall} />
            <ActionButton caption="2" action={() => action(deliveryOutcome, getScores(2))} noBall={noBall} />
            <ActionButton caption="3" action={() => action(deliveryOutcome, getScores(3))} noBall={noBall} />
            <ActionButton caption="4" action={() => action(deliveryOutcome, getScores(4))} noBall={noBall} />
            <OtherScore action={runs => action(deliveryOutcome, getScores(runs))}  noBall={noBall}/>
        </div>
    );
};
