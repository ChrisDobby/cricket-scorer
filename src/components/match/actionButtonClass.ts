import { DeliveryOutcome } from '../../domain';

const actionButtonClass = (invalidClassName: string) => (deliveryOutcome: DeliveryOutcome) =>
    deliveryOutcome === DeliveryOutcome.Noball || deliveryOutcome === DeliveryOutcome.Wide
        ? invalidClassName
        : undefined;

export default actionButtonClass('btn-danger');
