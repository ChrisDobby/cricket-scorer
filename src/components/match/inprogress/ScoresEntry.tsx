import * as React from 'react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';
import green from '@material-ui/core/colors/green';
import red from '@material-ui/core/colors/red';
import { DeliveryOutcome, DeliveryScores } from '../../../domain';
import executeDeliveryAction from './executeDeliveryAction';
import Tooltip from '../../Tooltip';
import { ScoreType } from './ScoreTypeSelect';
import HelpTooltip from '../../HelpTooltip';
import HelpContent from '../../HelpContent';

const buttonStyle = (deliveryOutcome: DeliveryOutcome): React.CSSProperties => ({
    backgroundColor: deliveryOutcome === DeliveryOutcome.Valid ? green[600] : red[700],
    marginRight: '8px',
    marginBottom: '8px',
    color: '#ffffff',
});

const boundary = (score: number) => (
    <span style={{ fontSize: '70%' }}>
        boundary
        <br />
        <span style={{ fontSize: '200%' }}>{score}</span>
    </span>
);

interface ScoresEntryProps {
    deliveryOutcome: DeliveryOutcome;
    scoreType: ScoreType;
    hasBoundaries: boolean;
    getScores: (score: number) => DeliveryScores;
    action: (deliveryOutcome: DeliveryOutcome, scores: DeliveryScores) => void;
}

const executeNonStandard = (score: number, execute: (score: number) => () => void) => {
    if (score >= 4) {
        execute(score)();
    }
};

const dotDescription = (deliveryOutcome: DeliveryOutcome) => {
    if (deliveryOutcome === DeliveryOutcome.Wide) {
        return 'Record a wide with no extra runs';
    }

    if (deliveryOutcome === DeliveryOutcome.Noball) {
        return 'Record a no ball with no extra runs';
    }

    return 'Record a dot ball';
};

const runsDescription = (deliveryOutcome: DeliveryOutcome, scoreType: ScoreType) => (
    runs: number,
    isBoundary?: boolean,
) => {
    const scoreDescription = () => {
        switch (scoreType) {
            case ScoreType.Runs:
                return `${runs === 1 ? '1 run' : `${runs} runs`}`;
            case ScoreType.Byes:
                return `${runs === 1 ? '1 bye' : `${runs} byes`}`;
            case ScoreType.LegByes:
                return `${runs === 1 ? '1 leg bye' : `${runs} leg byes`}`;
            default:
                return '';
        }
    };

    const addBoundary = (text: string) => (isBoundary ? `${text} (boundary)` : text);

    if (deliveryOutcome === DeliveryOutcome.Wide) {
        return addBoundary(`Record a wide with ${runs === 1 ? 'an extra run' : `${runs} extra runs`}`);
    }

    if (deliveryOutcome === DeliveryOutcome.Noball) {
        return addBoundary(`Record a no ball with ${scoreDescription()}`);
    }

    return addBoundary(scoreDescription());
};

export default ({ deliveryOutcome, scoreType, hasBoundaries, getScores, action }: ScoresEntryProps) => {
    const execute = executeDeliveryAction(action, getScores, deliveryOutcome);
    const executeBoundary = executeDeliveryAction(
        action,
        deliveryOutcome === DeliveryOutcome.Wide || scoreType !== ScoreType.Runs
            ? getScores
            : score => ({ boundaries: score }),
        deliveryOutcome,
    );

    const executeDot = executeDeliveryAction(action, () => ({ runs: 0 }), deliveryOutcome);
    const style = buttonStyle(deliveryOutcome);

    const runsTitle = runsDescription(deliveryOutcome, scoreType);
    return (
        <Grid container>
            <Grid container>
                <Tooltip title={dotDescription(deliveryOutcome)}>
                    <Button variant="fab" aria-label="Dot ball" style={style} onClick={executeDot(0)}>
                        {'.'}
                    </Button>
                </Tooltip>
                <Tooltip title={runsTitle(1)}>
                    <Button variant="fab" aria-label="Single" style={style} onClick={execute(1)}>
                        {'1'}
                    </Button>
                </Tooltip>
                <Tooltip title={runsTitle(2)}>
                    <Button variant="fab" aria-label="Two" style={style} onClick={execute(2)}>
                        {'2'}
                    </Button>
                </Tooltip>
                <Tooltip title={runsTitle(3)}>
                    <Button variant="fab" aria-label="Three" style={style} onClick={execute(3)}>
                        {'3'}
                    </Button>
                </Tooltip>
                <Tooltip title={runsTitle(4, true)}>
                    <Button variant="extendedFab" aria-label="Boundary Four" style={style} onClick={executeBoundary(4)}>
                        {boundary(4)}
                    </Button>
                </Tooltip>
                {hasBoundaries && (
                    <Tooltip title={runsTitle(6, true)}>
                        <Button
                            variant="extendedFab"
                            aria-label="Boundary Six"
                            style={style}
                            onClick={executeBoundary(6)}
                        >
                            {boundary(6)}
                        </Button>
                    </Tooltip>
                )}
            </Grid>
            <Grid container>
                <HelpTooltip title={<HelpContent.RunsScored />}>
                    <Select value={0} onChange={ev => executeNonStandard(Number(ev.target.value), execute)}>
                        <MenuItem value={0}>or select a score...</MenuItem>
                        <MenuItem value={4}>4</MenuItem>
                        <MenuItem value={5}>5</MenuItem>
                        <MenuItem value={6}>6</MenuItem>
                        <MenuItem value={7}>7</MenuItem>
                        <MenuItem value={8}>8</MenuItem>
                        <MenuItem value={9}>9</MenuItem>
                        <MenuItem value={10}>10</MenuItem>
                    </Select>
                </HelpTooltip>
            </Grid>
        </Grid>
    );
};
