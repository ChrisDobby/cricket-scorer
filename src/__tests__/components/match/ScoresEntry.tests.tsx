import * as React from 'react';
import * as ReactTestRenderer from 'react-test-renderer';
import { ScoresEntry } from '../../../components/match/ScoresEntry';
import { DeliveryOutcome } from '../../../domain';

describe('ScoresEntry', () => {
    it('should render correctly when valid delivery and showing dot', () => {
        const scoresEntry = ReactTestRenderer.create(
            <ScoresEntry
                showDot={true}
                deliveryOutcome={DeliveryOutcome.Valid}
                getScores={jest.fn()}
                action={jest.fn()}
            />);

        expect(scoresEntry.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when no ball', () => {
        const scoresEntry = ReactTestRenderer.create(
            <ScoresEntry
                showDot={true}
                deliveryOutcome={DeliveryOutcome.Noball}
                getScores={jest.fn()}
                action={jest.fn()}
            />);

        expect(scoresEntry.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when wide', () => {
        const scoresEntry = ReactTestRenderer.create(
            <ScoresEntry
                showDot={true}
                deliveryOutcome={DeliveryOutcome.Wide}
                getScores={jest.fn()}
                action={jest.fn()}
            />);

        expect(scoresEntry.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when valid delivery and not showing dot', () => {
        const scoresEntry = ReactTestRenderer.create(
            <ScoresEntry
                showDot={false}
                deliveryOutcome={DeliveryOutcome.Valid}
                getScores={jest.fn()}
                action={jest.fn()}
            />);

        expect(scoresEntry.toJSON()).toMatchSnapshot();
    });
});
