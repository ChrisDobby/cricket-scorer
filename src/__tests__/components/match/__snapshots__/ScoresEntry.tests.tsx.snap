// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoresEntry should render correctly when no ball 1`] = `
<div
  className="col-10"
>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": "50%",
          "borderStyle": "solid",
          "borderWidth": "1px",
        }
      }
    >
      .
    </div>
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": "50%",
          "borderStyle": "solid",
          "borderWidth": "1px",
        }
      }
    >
      1
    </div>
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": "50%",
          "borderStyle": "solid",
          "borderWidth": "1px",
        }
      }
    >
      2
    </div>
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": "50%",
          "borderStyle": "solid",
          "borderWidth": "1px",
        }
      }
    >
      3
    </div>
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
        "paddingLeft": "4px",
        "paddingRight": "4px",
      }
    }
  >
    <div
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": "50%",
          "borderStyle": "solid",
          "borderWidth": "1px",
        }
      }
    >
      4
    </div>
  </button>
  <span
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  >
    <select
      className="custom-select"
      onChange={[Function]}
      style={
        Object {
          "width": "84px",
        }
      }
      value="other"
    >
      <option>
        other
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
    <button
      className="btn btn-danger"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
    >
      <i
        className="fa fa-check"
      />
    </button>
  </span>
</div>
`;

exports[`ScoresEntry should render correctly when valid delivery and not showing dot 1`] = `
<div
  className="col-10"
>
  <span
    style={
      Object {
        "display": "inline-block",
        "width": "40px",
      }
    }
  />
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    1
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    2
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    3
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    4
  </button>
  <span
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  >
    <select
      className="custom-select"
      onChange={[Function]}
      style={
        Object {
          "width": "84px",
        }
      }
      value="other"
    >
      <option>
        other
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
    <button
      className="btn btn-success"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
    >
      <i
        className="fa fa-check"
      />
    </button>
  </span>
</div>
`;

exports[`ScoresEntry should render correctly when valid delivery and showing dot 1`] = `
<div
  className="col-10"
>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    .
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    1
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    2
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    3
  </button>
  <button
    className="btn btn-success"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    4
  </button>
  <span
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  >
    <select
      className="custom-select"
      onChange={[Function]}
      style={
        Object {
          "width": "84px",
        }
      }
      value="other"
    >
      <option>
        other
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
    <button
      className="btn btn-success"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
    >
      <i
        className="fa fa-check"
      />
    </button>
  </span>
</div>
`;

exports[`ScoresEntry should render correctly when wide 1`] = `
<div
  className="col-10"
>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    .
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    1
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    2
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    3
  </button>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "2px",
        "minWidth": "36px",
      }
    }
  >
    4
  </button>
  <span
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  >
    <select
      className="custom-select"
      onChange={[Function]}
      style={
        Object {
          "width": "84px",
        }
      }
      value="other"
    >
      <option>
        other
      </option>
      <option
        value="5"
      >
        5
      </option>
      <option
        value="6"
      >
        6
      </option>
      <option
        value="7"
      >
        7
      </option>
      <option
        value="8"
      >
        8
      </option>
      <option
        value="9"
      >
        9
      </option>
      <option
        value="10"
      >
        10
      </option>
    </select>
    <button
      className="btn btn-success"
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "margin": "2px",
        }
      }
    >
      <i
        className="fa fa-check"
      />
    </button>
  </span>
</div>
`;
